### ISSUE TITLE
Eager queries of relations within embedded entities do not work

### ISSUE THREAD

**Issue by awrichar** [nestjsx/crud#669]
Given an entity which contains an embedded entity, which in turn contains a relation:
```typescript
@Entity()
export class Phone {
  @PrimaryGeneratedColumn() id: number;
  @Column() number: string;
}

export class ContactDetails {
  @ManyToMany(() => Phone, {cascade: true})
  @JoinTable()
  phone: Phone[];
}

@Entity()
export class Event {
  @PrimaryGeneratedColumn() id: number;
  @Column(() => ContactDetails) contactDetails: ContactDetails
}
```

I would like for my query to include all nested phone entries by default. Neither of these `join` options works:
```typescript
@Crud({
  model: {type: Event},
  query: {
    join: {
      'phone': {eager: true},  // Error: Relation with property path phone in entity was not found.
      'contactDetails.phone': {eager: true},  // Silently fails (does not include phone info)
    },
  },
})
@Controller('event')
export class EventController implements CrudController<Event> {
  constructor(public service: EventService) {}
}
```

I'm able to hack it into working by adding this to my service initialization (caching the relation and modifying the path to include the embedded entity name):
```typescript
@Injectable()
export class EventService extends TypeOrmCrudService<Event> {
  constructor(
    @InjectRepository(Event) repo: Repository<Event>
  ) {
    super(repo);
    let phone = this.getRelationMetadata('phone', {});
    phone.path = 'Event.contactDetails.phone';
  }
}
```

After that, I can set `query: { join: { phone: {eager: true} } }` on my controller and it will properly include the phone info in my queries. Am I missing any way to make this work without the hack above?

Seems like possibly a bug with how relations in embedded entities are addressed (ie an inconsistency in whether the path includes the embedded entity or not).

**Comment by zMotivat0r** [comment]
Have you tried to set an alias for your relations in `query.join` options?
Please try it and also I would suggest taking a look at some closed issues.

**Comment by awrichar** [comment]
@zMotivat0r yes, I did find #512 first, which deals with nested JOINs (ie a relation within a relation). I actually had that same problem elsewhere, and adding aliases in that case solves the issue.

This case is different, since it's only a single JOIN - but because the field being joined is within an embedded entity, the naming of the relation is odd, and seems to be handled differently by `getRelationMetadata` and `leftJoin`. I tried a few things with aliases and they do not help in this case. The relation `path` cached by `getRelationMetadata` is always incorrect and causes the actual join to fail.

Upon searching again this morning, this looks like a duplicate of #533. My "hack fix" is a bit smaller, but I think we've both keyed in to the fact that something is off in `getRelationMetadata` for this case.

**Comment by awrichar** [comment]
As far as I can tell, the problem is with the logic in `getRelationMetadata` that builds the path to the nested relation:
```
                        if (i !== fields.length - 1) {
                            parentPath = !parentPath
                                ? propertyName
                                : `${parentPath}.${propertyName}`;
                        }
```

Specifically concatenating `${parentPath}.${propertyName}` causes the introduction of the invalid join path. Possibly this should be `${parentPath}.${propertyPath}`? I'm unsure if that has other breaking side-effects; I haven't had the time to dig in too far.

**Comment by awrichar** [comment]
This later path building logic is problematic as well:
```
if (!path && parentPath) {
    const parentAllowedRelation = this.entityRelationsHash.get(parentPath);
    if (parentAllowedRelation) {
        path = parentAllowedRelation.alias
            ? `${parentAllowedRelation.alias}.${name}`
            : field;
    }
}
```

That one is slightly more complicated, since the `found` item for the relation isn't in scope. But again, it seems like this logic is assuming that the "name" and "path" of a relation are the same, which doesn't hold up for relations in embedded entities (ie when propertyName is "phone" and propertyPath is "contactDetails.phone", the latter should be used when building the path).
